### PR TITLE
Don't count (connecting) players for voting

### PR DESCRIPTION
**Pull request by def-**
## Checklist

- [ ] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

File: src/game/server/gamecontext.cpp [hunk 1]
```diff
@@ -1507,7 +1507,8 @@ void CGameContext::OnClientConnected(int ClientID, void *pData)
 		bool Empty = true;
 		for(auto &pPlayer : m_apPlayers)
 		{
-			if(pPlayer)
+			// connecting clients with spoofed ips can clog slots without being ingame
+			if(pPlayer && ((CServer *)Server())->m_aClients[pPlayer->GetCID()].m_State != CServer::CClient::STATE_INGAME)
```
Comment (Robyt3) [inline src/game/server/gamecontext.cpp:1511]: `!Server()->ClientIngame(pPlayer->GetCID())` then you don't need the cast.

Comment (def-) [reply src/game/server/gamecontext.cpp]: Done, thanks

Comment (heinrich5991) [reply src/game/server/gamecontext.cpp]: Generally, this cast should never be done. It breaks the engine-game abstraction.

**Pull request merged**